src/libertem/udf/base.py

```diff
@@ -2581,15 +2581,19 @@ def _inner():
                     yield params_handle
 
                     if tasks:
-                        for res in executor.run_tasks(
+                        task_gen = executor.run_tasks(
                             tasks,
                             params_handle,
                             cancel_id,
                             task_comm_handler,
-                        ):
-                            if progress:
-                                pman.finalize_task(res[1])
-                            yield res
+                        )
+                        try:
+                            for res in task_gen:
+                                if progress:
+                                    pman.finalize_task(res[1])
```

Comment:
I'm not sure why this isn't covered, even with the additional test case; `progress` should be passed down properly, and the next line has coverage...

Reply:
When I add a `raise Exception` before that line in the same `if` statement, `tests/executor/test_pipelined.py::test_with_progress` fails exactly there. I'd say it is definitely covered provided that test runs in CI (and why shouldn't it), and Codecov is mistaken.